### PR TITLE
CLI Peers Management

### PR DESCRIPTION
**Pull request by tiram88**
Enhance `kaspad` CLI with following repeatable arguments:
- `--connect=` and `--addpeer=` for peers management.
- `--uacomment=` for user-agent comments describing the launched node (see [BIP-014](https://en.bitcoin.it/wiki/BIP_0014) for more information)

### CODE REVIEW

File: kaspad/src/args.rs [hunk 1]
```diff
@@ -111,23 +117,37 @@ pub fn cli(defaults: &Defaults) -> Command {
...
         )
         .arg(arg!(--unsaferpc "Enable RPC commands which affect the state of the node"))
         .arg(
-            Arg::new("connect")
+            Arg::new("connect-peers")
                 .long("connect")
                 .value_name("IP[:PORT]")
-                .num_args(0..=1)
+                .action(ArgAction::Append)
+                .num_args(0..=10) // TODO: define an upper bound
```
Comment (michaelsutton) [inline kaspad/src/args.rs:129]: Are you sure `num_args` is relevant when using `ArgAction::Append` ? we should manually test it.

File: kaspad/src/main.rs [hunk 2]
```diff
@@ -188,7 +203,10 @@ pub fn main() {
     // DB used for addresses store and for multi-consensus management
     let meta_db = kaspa_database::prelude::open_db(meta_db_dir, true, 1);
 
-    let grpc_server_addr = args.rpclisten.unwrap_or_else(|| format!("0.0.0.0:{}", config.default_rpc_port())).parse().unwrap();
+    let connect_peers = args.connect_peers.iter().map(|x| x.normalize(config.default_p2p_port())).collect();
+    let add_peers = args.add_peers.iter().map(|x| x.normalize(config.default_p2p_port())).collect();
+    let listen = args.listen.unwrap_or_default().normalize(config.default_rpc_port());
```
Comment (michaelsutton) [inline kaspad/src/main.rs:208]: `default_rpc_port` -> `default_p2p_port`. Also probably better to rename `listen` -> `p2p_server_addr`

File: protocol/flows/src/service.rs [hunk 3]
```diff
@@ -71,8 +75,11 @@ impl AsyncService for P2pService {
 
         // Launch the service and wait for a shutdown signal
         Box::pin(async move {
-            if let Some(peer_address) = self.connect.clone() {
-                connection_manager.add_connection_request(peer_address.to_socket_addrs().unwrap().next().unwrap(), true).await;
+            for peer_address in self.connect_peers.iter().cloned() {
```
Comment (michaelsutton) [inline protocol/flows/src/service.rs:78]: I think in go-kaspad having `connect` means exclusiveness. Meaning the node does not attempt any other outbound connections (can be achieved by setting `outbound_target=0` if `!self.connect_peers.is_empty()`. I'm not sure about inbound connections

File: kaspad/src/args.rs (hunk 1 above)
Comment (michaelsutton) [reply kaspad/src/args.rs:129]: In general I'm not sure I understand the role of `.num_args(0..=1)` in all other flags

Comment (tiram88) [reply kaspad/src/args.rs]: I agree on both remarks.

I removed all `.num_args(iter)` and tested manually. The system exhibits the desired behavior as expected.

File: kaspad/src/main.rs (hunk 2 above)
Comment (tiram88) [reply kaspad/src/main.rs:208]: Nice catch!

File: protocol/flows/src/service.rs (hunk 3 above)
Comment (tiram88) [reply protocol/flows/src/service.rs]: Right, this exclusiveness is achieved in go-kaspad by setting outbound target to 0 and disabling DNS seeding. I added that.

I could find no impact of the presence of `connect` in go-kaspad on inbound connections.

I also discovered and fixed another little discrepancy: added peers have permanent connections.

**Review by michaelsutton — APPROVED** [review]

**Pull request merged**